sourcegraph/sourcegraph
# Cody: Fix some Non-Stop Fixup TODOs in state transitions

Moves the transitions from the FixupTask into the controller. With multiple tasks, it does not make sense to put transition side effects like setting VSCode context, in a single task. This needs to be handled by the controller which has oversight of all the tasks.

Fixes a TODO where we repeatedly set the id &rarr; task map on every state transition instead of once at startup.

Removes the editor from the FixupTask. Multiple editors can open the same file, so it's not ideal to hold onto any one editor. Instead we should use the best editor available at any given time when we need an editor.

## Test plan

```
pnpm test:unit && pnpm test:integration && pnpm test:e2e
```

Manual test:

1. Open a file.
2. Cmd-Shift-P Cody: Fixup (Experimental)
3. Tell Cody to do some refactoring.
4. Check blue highlights appear, change can be applied, etc.
